src/nvim/ex_getln.c

```diff
@@ -1669,7 +1672,16 @@ getcmdline (
     int indent               // indent for inside conditionals
 )
 {
-  return command_line_enter(firstc, count, indent);
+  if (ui_is_external(kUICmdline)) {
+    Array args = ARRAY_DICT_INIT;
+    ui_event("cmdline_enter", args);
```

Comment:
I don't think the ui is really supposed to keep track of the mode stack-machine, it should just be told what to draw right now. When entering the sub cmdline it will see `cmdline_show` with the new relevant attributes. When leaving, it will see `cmdline_hide` immediately followed by `cmdline_show` for the outer command line. In the end it will always see consistent state.